### ISSUE TITLE
Customized Metadata at Datafile Level

### ISSUE THREAD

**Issue by lincolnsherpa** [IQSS/dataverse#7930]
I have a Usecase where an experiment contains a group of images /frames and each frame has a unique metadata field. They are ultimately intended for ML application.
Currently, Dataverse has limitation in datafile metadata fields. It would be awesome to have customized metadata fields at the datafile level.

The concept is taken from Matlab. (https://de.mathworks.com/help/matlab/datastore.html)

Not exactly 100% be almost similar.

**Comment by pdurbin** [comment]
@lincolnsherpa yes, this is a fairly common request to have additional metadata at the file level. See, for example:

- #916
- #2295
- #6455
